src/processing/script/transform.js

```diff
@@ -21,33 +22,64 @@ function getChange (node, parent, key) {
     };
 }
 
-function transformChildNodes (node, changes, parentChanged) {
+// NOTE: There is an issue with processing `new` expressions. `new a.src.b()` will be transformed
+// to `new __get$(a, 'src').b()`, which is wrong. The correct result is `new (__get$(a, 'src')).b()`.
+// To solve this problem, we add a 'state' entity. This entity stores the "new" expression, so that
+// we can add it to the changes when the transformation is found.
+function createState (currState, node, parent, key, hasTransformedAncestor) {
+    var isNewExpression         = node.type === Syntax.NewExpression;
+    var isNewExpressionAncestor = isNewExpression && !currState.newExpressionAncestor;
+
+    return {
+        hasTransformedAncestor:      currState.hasTransformedAncestor || hasTransformedAncestor,
+        newExpressionAncestor:       isNewExpressionAncestor ? node : currState.newExpressionAncestor,
+        newExpressionAncestorParent: isNewExpressionAncestor ? parent : currState.newExpressionAncestorParent,
+        newExpressionAncestorKey:    isNewExpressionAncestor ? key : currState.newExpressionAncestorKey
+    };
+}
+
+function transformChildNodes (node, changes, state) {
     for (var key in node) {
         if (node.hasOwnProperty(key)) {
             var childNode = node[key];
 
             if (objectToString.call(childNode) === '[object Array]') {
                 for (var j = 0; j < childNode.length; j++)
-                    transform(childNode[j], node, key, changes, parentChanged);
+                    transform(childNode[j], node, key, changes, state);
             }
             else
-                transform(childNode, node, key, changes, parentChanged);
+                transform(childNode, node, key, changes, state);
         }
     }
 }
 
-export default function transform (node, parent, key, changes, parentChanged) {
-    var nodeChanged = false;
+function isNodeTransformed (node) {
+    return node.originStart !== void 0 && node.originEnd !== void 0;
+}
 
+function addChangeForTransformedNode (state, changes, replacement, parent, key) {
+    if (!state.hasTransformedAncestor &&
+        (!state.newExpressionAncestor || !isNodeTransformed(state.newExpressionAncestor))) {
```

Comment:
Please, avoid such multiline conditions. Keep it single line or even better refactor it to the var with meaningful name